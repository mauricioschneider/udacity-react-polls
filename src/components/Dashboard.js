import { useState } from "react";
import { connect } from "react-redux";
import PollsSection from "./PollsSection";

function classNames(...classes) {
  return classes.filter(Boolean).join(" ");
}

const Dashboard = () => {
  const [type, setType] = useState("new");

  const handleTypeSwitch = (e) => {
    e.preventDefault();

    setType(e.target.value);
  };
  return (
    <div className="min-h-full">
      <main>
        <div className="mx-auto max-w-7xl sm:px-6 lg:px-8">
          <div className="md:flex md:items-center md:justify-between">
            <div className="min-w-0 p-4 flex-1">
              <h2 className="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl sm:tracking-tight">
                All Polls
              </h2>
            </div>
            <small className="">Filter:</small>
            <div className="mt-4 flex md:ml-4 md:mt-0">
              <button
                type="button"
                onClick={handleTypeSwitch}
                value="new"
                className={classNames(
                  type === "new"
                    ? "bg-indigo-600 text-white hover:bg-indigo-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    : "bg-white hover:bg-gray-50",
                  "inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                )}
              >
                New Questions
              </button>
              <button
                type="button"
                onClick={handleTypeSwitch}
                value="done"
                className={classNames(
                  type === "done"
                    ? "bg-indigo-600 text-white hover:bg-indigo-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                    : "bg-white hover:bg-gray-50",
                  "ml-3 inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                )}
              >
                Done
              </button>
            </div>
          </div>

          <PollsSection type={type} />
        </div>
      </main>
    </div>
  );
};

export default connect()(Dashboard);
