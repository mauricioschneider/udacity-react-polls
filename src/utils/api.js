import {
  _getUsers,
  _getQuestions,
  _saveQuestionAnswer,
  _saveQuestion,
} from "./_DATA";

export function getInitialData() {
  return Promise.all([_getUsers(), _getQuestions()]).then(([users, polls]) => ({
    users,
    polls,
  }));
}

export function saveQuestionAnswer(info) {
  // info = { authedUser, qid, answer }
  return _saveQuestionAnswer(info);
}

export function saveQuestion(question) {
  // question = {optionOneText, optionTwoText, author}
  console.log(`question: `, question);
  return _saveQuestion(question);
}
