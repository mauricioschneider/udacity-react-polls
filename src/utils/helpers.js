export function formatDate(timestamp) {
  const d = new Date(timestamp);
  const time = d.toLocaleTimeString("en-US");
  return time.substr(0, 5) + time.slice(-2) + " | " + d.toLocaleDateString();
}

export function formatPoll(poll, author, authedUser) {
  const { id, timestamp } = poll;
  const { name, avatarURL } = author;

  const allVotes = poll.optionOne.votes.concat(poll.optionTwo.votes);
  const hasVoted = allVotes.includes(authedUser);

  return {
    name,
    id,
    timestamp,
    avatar: avatarURL,
    votes: allVotes.length,
    hasVoted,
    type: hasVoted ? "done" : "new",
    options: [poll.optionOne, poll.optionTwo],
  };
}
