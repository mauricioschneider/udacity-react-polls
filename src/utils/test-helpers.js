import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import reducer from "../reducers";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MemoryRouter>{children}</MemoryRouter>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
