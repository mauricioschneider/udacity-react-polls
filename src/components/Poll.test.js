import { renderWithProviders } from "../utils/test-helpers";

import Dashboard from "./Dashboard";

describe("The Dashboard component", () => {
  test("shows list of polls", () => {
    expect(renderWithProviders(<Dashboard />)).toMatchSnapshot();
  });
});
