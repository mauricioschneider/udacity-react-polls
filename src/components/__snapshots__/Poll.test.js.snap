// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Dashboard component shows list of polls 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-h-full"
      >
        <main>
          <div
            class="mx-auto max-w-7xl sm:px-6 lg:px-8"
          >
            <div
              class="md:flex md:items-center md:justify-between"
            >
              <div
                class="min-w-0 p-4 flex-1"
              >
                <h2
                  class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl sm:tracking-tight"
                >
                  All Polls
                </h2>
              </div>
              <small
                class=""
              >
                Filter:
              </small>
              <div
                class="mt-4 flex md:ml-4 md:mt-0"
              >
                <button
                  class="bg-indigo-600 text-white hover:bg-indigo-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                  type="button"
                  value="new"
                >
                  New Questions
                </button>
                <button
                  class="bg-white hover:bg-gray-50 ml-3 inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                  type="button"
                  value="done"
                >
                  Done
                </button>
              </div>
            </div>
            <div
              class="bg-gray-100 py-8"
            >
              <div
                class="mx-auto max-w-7xl sm:px-6 lg:px-8"
              >
                <ul
                  class="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3"
                />
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-h-full"
    >
      <main>
        <div
          class="mx-auto max-w-7xl sm:px-6 lg:px-8"
        >
          <div
            class="md:flex md:items-center md:justify-between"
          >
            <div
              class="min-w-0 p-4 flex-1"
            >
              <h2
                class="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl sm:tracking-tight"
              >
                All Polls
              </h2>
            </div>
            <small
              class=""
            >
              Filter:
            </small>
            <div
              class="mt-4 flex md:ml-4 md:mt-0"
            >
              <button
                class="bg-indigo-600 text-white hover:bg-indigo-700 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600 inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                type="button"
                value="new"
              >
                New Questions
              </button>
              <button
                class="bg-white hover:bg-gray-50 ml-3 inline-flex items-center rounded-md px-3 py-2 text-sm font-semibold text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300"
                type="button"
                value="done"
              >
                Done
              </button>
            </div>
          </div>
          <div
            class="bg-gray-100 py-8"
          >
            <div
              class="mx-auto max-w-7xl sm:px-6 lg:px-8"
            >
              <ul
                class="grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3"
              />
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
